sympy/core/expr.py

```diff
@@ -491,6 +495,18 @@ def is_constant(self, *wrt, **flags):
         True
         """
 
+        def check_denominator_zeros(expression):
+
+            from ..solvers import solveset
+            from ..solvers.solvers import denoms
+
+            for den in denoms(expression):
+                if (den.is_zero or
+                    any(solveset(den, symb) is not S.EmptySet
```

Comment:
I am -1 on using the expensive solve to answer the question. If the `is_zero` is None return None from the function.

Reply:
This routine wouldn't. If there is a question about a denominator that could be zero then the routine just doesn't answer, indicating to the user that some care should be taken in proceeding with this expression.